### PR TITLE
Add Haploflow

### PR DESCRIPTION
**Pull request by thomasbtf**
This PR adds the recipe for Haploflow. Haploflow is a strain-aware viral genome assembler for short read sequence data. [Click here](https://github.com/hzi-bifo/Haploflow) for further details.

Thanks @mbargull for the support!

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/haploflow/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    - boost-cpp
+
+
+test:
+  commands:
+    - haploflow --help
+
+about:
+  home: https://github.com/hzi-bifo/Haploflow
+  license: GPL-3.0 License
```
Comment (mbargull) [inline recipes/haploflow/meta.yaml:34]: I've opened https://github.com/hzi-bifo/Haploflow/issues/8 to ask whether `GPL-3.0-only` or something else is the intended license here.

Comment (mbargull) [reply recipes/haploflow/meta.yaml:34]: I got some clarifications from upstream:

> \[...\] I was waiting for the software to be published in a peer-reviewed journal to create a 1.0 release. Before that, v0.1 is probably the better version to use.

So, it's probably better to use the `v0.1` tag for now.
And
> \[...\] v0.1 used Apache and I updated the README.md when we changed to GPL for v0.2, but forgot to also update it to 0.2 in the text \[...\]

So, if you change this to use `v0.1`, then use
```suggestion
  license: Apache-2.0 AND MIT
```
here.

**Review by mbargull — APPROVED** [review]

**Comment by mbargull** [comment]
Oh, this is still marked as a "draft". From my part this looks perfectly ready to go. "Ready for review", @thomasbtf?

**Comment by mbargull** [comment]
@BiocondaBot, please merge

**Comment by thomasbtf** [comment]
Sure, didn't notice, that the CI passed

**Pull request merged**